src/components/WalletAssets.vue

```diff
@@ -207,8 +204,44 @@ export default class WalletAssets extends Mixins(LoadingMixin, FormattedAmountMi
     }
   }
 
-  mounted(): void {
-    this.assetList = api.assets.accountAssets;
+  filterAssets(filters: WalletAssetFilters): void {
+    if (filters.option === WalletFilteringOptions.ALL) {
+      this.assetList = api.assets.accountAssets.filter((asset) => this.applyFilterOptions(asset, filters));
```

Comment:
you use the same code 3 times
`arr.filter((asset) => this.applyFilterOptions(asset, filters)`